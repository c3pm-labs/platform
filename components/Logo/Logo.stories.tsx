import Logo, { LogoProps } from '.';

export default {
  title: 'Logo',
  component: Logo,
};

export const Basic = (args: LogoProps): JSX.Element => (<Logo {...args} />);
