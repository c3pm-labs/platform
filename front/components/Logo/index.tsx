import { Typography } from '@material-ui/core';
import makeStyles from '@material-ui/core/styles/makeStyles';
import Link from 'next/link';
import Image from 'next/image';
import { useTheme } from '@material-ui/core/styles';

function getFontSize(size: string): number {
  if (size === 'xl') {
    return 30;
  } if (size === 'lg') {
    return 20;
  } if (size === 'md') {
    return 16;
  }
  return 10;
}

function getSize(type: string, size: string): { width: number | string; height: number | string} {
  if (type === 'mini') {
    return ({ width: 30, height: 40 });
  }

  if (size === 'xl') {
    return ({ width: 600, height: 200 });
  }
  if (size === 'lg') {
    return ({ width: 300, height: 100 });
  }
  if (size === 'md') {
    return ({ width: 200, height: 35 });
  }
  if (size === 'sm') {
    return ({ width: 100, height: 35 });
  }
  return ({ width: '100%', height: 'auto' });
}

const useStyles = makeStyles((theme) => ({
  link: {
    textDecoration: 'none',
  },
  baseline: {
    color: theme.palette.primary.main,
    fontWeight: 'bold',
    fontSize: (props: { size: string}): number => getFontSize(props.size),
    textDecoration: 'none',
  },
  logo: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

export interface LogoProps {
  type?: 'classic' | 'mini' | 'baseline';
  size?: 'sm' | 'md' | 'lg' | 'xl' | null;
  className?: string;
  theme?: 'light' | 'dark'
}

const Logo = ({
  type = 'classic', size = null, className, theme = 'light',
}: LogoProps): JSX.Element => {
  const classes = useStyles({ size });
  const chooseLogo = () => {
    const src = type === 'mini' ? '/assets/c3pm_3' : '/assets/c3pm';

    return theme === 'light' ? `${src}.png` : `${src}_white.png`;
  };
  const src = chooseLogo();
  const dimension = getSize(type, size);

  return (
    <Link href="/">
      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
      <a className={classes.link}>
        <div className={classes.logo}>
          <Image
            alt={`${type}-${size} c3pm logo`}
            src={src}
            className={className}
            width={dimension.width}
            height={dimension.height}
          />
          { type === 'baseline' && (<Typography className={classes.baseline}>C++ Package Manager</Typography>)}
        </div>
      </a>
    </Link>
  );
};

export default Logo;
