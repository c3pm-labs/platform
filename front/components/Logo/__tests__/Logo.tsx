import { render } from '@testing-library/react';

import Logo from '../index';

test('Logo baseline xl', () => {
  const { container } = render(<Logo type="baseline" size="xl" />);

  expect(container).toMatchSnapshot();
});

test('Logo baseline md', () => {
  const { container } = render(<Logo type="baseline" size="md" />);

  expect(container).toMatchSnapshot();
});

test('Logo baseline lg', () => {
  const { container } = render(<Logo type="baseline" size="lg" />);

  expect(container).toMatchSnapshot();
});

test('Logo baseline sm', () => {
  const { container } = render(<Logo type="baseline" size="sm" />);

  expect(container).toMatchSnapshot();
});

test('Logo no size no type', () => {
  const { container } = render(<Logo />);

  expect(container).toMatchSnapshot();
});

test('Logo mini xl', () => {
  const { container } = render(<Logo type="mini" size="xl" />);

  expect(container).toMatchSnapshot();
});

test('Logo mini md', () => {
  const { container } = render(<Logo type="mini" size="md" />);

  expect(container).toMatchSnapshot();
});

test('Logo mini lg', () => {
  const { container } = render(<Logo type="mini" size="lg" />);

  expect(container).toMatchSnapshot();
});

test('Logo mini sm', () => {
  const { container } = render(<Logo type="mini" size="sm" />);

  expect(container).toMatchSnapshot();
});

test('Logo mini no size', () => {
  const { container } = render(<Logo type="mini" />);

  expect(container).toMatchSnapshot();
});
