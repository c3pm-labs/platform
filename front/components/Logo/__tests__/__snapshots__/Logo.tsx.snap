// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo baseline lg 1`] = `
<div>
  <a
    class="makeStyles-link-9"
    href="/"
  >
    <div
      class="makeStyles-logo-11"
    >
      <img
        alt="baseline-lg c3pm logo"
        height="auto"
        src="/assets/c3pm.png"
        width="300"
      />
      <p
        class="MuiTypography-root makeStyles-baseline-10 makeStyles-baseline-12 MuiTypography-body1"
      >
        C++ Package Manager
      </p>
    </div>
  </a>
</div>
`;

exports[`Logo baseline md 1`] = `
<div>
  <a
    class="makeStyles-link-5"
    href="/"
  >
    <div
      class="makeStyles-logo-7"
    >
      <img
        alt="baseline-md c3pm logo"
        height="auto"
        src="/assets/c3pm.png"
        width="200"
      />
      <p
        class="MuiTypography-root makeStyles-baseline-6 makeStyles-baseline-8 MuiTypography-body1"
      >
        C++ Package Manager
      </p>
    </div>
  </a>
</div>
`;

exports[`Logo baseline sm 1`] = `
<div>
  <a
    class="makeStyles-link-13"
    href="/"
  >
    <div
      class="makeStyles-logo-15"
    >
      <img
        alt="baseline-sm c3pm logo"
        height="auto"
        src="/assets/c3pm.png"
        width="100"
      />
      <p
        class="MuiTypography-root makeStyles-baseline-14 makeStyles-baseline-16 MuiTypography-body1"
      >
        C++ Package Manager
      </p>
    </div>
  </a>
</div>
`;

exports[`Logo baseline xl 1`] = `
<div>
  <a
    class="makeStyles-link-1"
    href="/"
  >
    <div
      class="makeStyles-logo-3"
    >
      <img
        alt="baseline-xl c3pm logo"
        height="auto"
        src="/assets/c3pm.png"
        width="600"
      />
      <p
        class="MuiTypography-root makeStyles-baseline-2 makeStyles-baseline-4 MuiTypography-body1"
      >
        C++ Package Manager
      </p>
    </div>
  </a>
</div>
`;

exports[`Logo mini lg 1`] = `
<div>
  <a
    class="makeStyles-link-29"
    href="/"
  >
    <div
      class="makeStyles-logo-31"
    >
      <img
        alt="mini-lg c3pm logo"
        height="100"
        src="/assets/c3pm_3.png"
        width="auto"
      />
    </div>
  </a>
</div>
`;

exports[`Logo mini md 1`] = `
<div>
  <a
    class="makeStyles-link-25"
    href="/"
  >
    <div
      class="makeStyles-logo-27"
    >
      <img
        alt="mini-md c3pm logo"
        height="80"
        src="/assets/c3pm_3.png"
        width="auto"
      />
    </div>
  </a>
</div>
`;

exports[`Logo mini no size 1`] = `
<div>
  <a
    class="makeStyles-link-37"
    href="/"
  >
    <div
      class="makeStyles-logo-39"
    >
      <img
        alt="mini-null c3pm logo"
        height="auto"
        src="/assets/c3pm_3.png"
        width="auto"
      />
    </div>
  </a>
</div>
`;

exports[`Logo mini sm 1`] = `
<div>
  <a
    class="makeStyles-link-33"
    href="/"
  >
    <div
      class="makeStyles-logo-35"
    >
      <img
        alt="mini-sm c3pm logo"
        height="40"
        src="/assets/c3pm_3.png"
        width="auto"
      />
    </div>
  </a>
</div>
`;

exports[`Logo mini xl 1`] = `
<div>
  <a
    class="makeStyles-link-21"
    href="/"
  >
    <div
      class="makeStyles-logo-23"
    >
      <img
        alt="mini-xl c3pm logo"
        height="200"
        src="/assets/c3pm_3.png"
        width="auto"
      />
    </div>
  </a>
</div>
`;

exports[`Logo no size no type 1`] = `
<div>
  <a
    class="makeStyles-link-17"
    href="/"
  >
    <div
      class="makeStyles-logo-19"
    >
      <img
        alt="classic-null c3pm logo"
        height="auto"
        src="/assets/c3pm.png"
        width="100%"
      />
    </div>
  </a>
</div>
`;
